src/sidebars/star/star-controller.js

```diff
@@ -15,33 +15,35 @@
             this.selectionService = selectionService;
 
             this.starHovered = false;
-            this.check = false;
+
+            this.stock = $scope.stock;
```

Comment:
It looks like the `$scope` variable was a hangover from before... perhaps it doesn't need to be assigned to `this.$scope` above? (I know that's not strictly part of this refactor though.)